llama-cli/src/main.rs

```diff
@@ -35,12 +37,51 @@ fn main() {
         std::process::exit(1);
     };
 
-    let (model, vocab) = llama_rs::Model::load(&args.model_path, args.num_ctx_tokens as i32)
+    let (mut model, vocab) = llama_rs::Model::load(&args.model_path, args.num_ctx_tokens as i32)
         .expect("Could not load model");
 
+    if let Some(restore_path) = &args.restore_prompt {
+        let memory = ModelMemory::load_compressed(restore_path);
```

Comment:
You can load a stored memory file from disk like this. I chose to add a simple API with `bincode`, but the struct implements `De`/`Serialize` so you can do your own serialization instead.